src/documents/models.py

```diff
@@ -292,12 +293,33 @@ def file_type(self):
         return get_default_file_extension(self.mime_type)
 
     @property
-    def thumbnail_path(self):
-        file_name = f"{self.pk:07}.png"
+    def thumbnail_path(self) -> str:
+        png_file_name = f"{self.pk:07}.png"
+        webp_file_name = f"{self.pk:07}.webp"
         if self.storage_type == self.STORAGE_TYPE_GPG:
-            file_name += ".gpg"
-
-        return os.path.join(settings.THUMBNAIL_DIR, file_name)
+            png_file_name += ".gpg"
+            webp_file_name += ".gpg"
+
+        # This property is used to both generate the file path
+        # and locate the file itself
+        # Hence why this looks a little weird
+
+        webp_file_path = os.path.join(settings.THUMBNAIL_DIR, webp_file_name)
+        png_file_path = os.path.join(settings.THUMBNAIL_DIR, png_file_name)
+
+        # 1. Assume the thumbnail is WebP
```

Comment:
I think this whole block (310-321) can be replaced by for less file checks, readability, less nested `if`s, and less `not if` logics:
```
thumb = webp_file_path
if os.path.exists(png_file_path):
    thumb = png_file_path
```

it seems the `if not os.path.exists(webp_file_path):` check is irrelevant.

We *only* set the PNG thumb path if that file exists, no need to check if webp exists